src/pyupgrade/fixes.rs

```diff
@@ -133,6 +137,78 @@ pub fn remove_super_arguments(locator: &SourceCodeLocator, expr: &Expr) -> Optio
     None
 }
 
+/// U010
+/// This fix mirrors the pyflakes::fixes::remove_unused_import_froms, possible
+/// merger of common code into helpers is better
+pub fn remove_unnecessary_future_import(
+    locator: &SourceCodeLocator,
+    stmt: &Stmt,
+    version: PythonVersion,
+    names: &[Located<AliasData>],
+) -> Result<Fix> {
+    let module_text = locator.slice_source_code_range(&Range::from_located(stmt));
+    let mut tree = match_module(&module_text)?;
+
+    let body = if let Some(Statement::Simple(body)) = tree.body.first_mut() {
+        body
+    } else {
+        return Err(anyhow::anyhow!("Expected node to be: Statement::Simple"));
+    };
+    let body = if let Some(SmallStatement::ImportFrom(body)) = body.body.first_mut() {
+        body
+    } else {
+        return Err(anyhow::anyhow!(
+            "Expected node to be: SmallStatement::ImportFrom"
+        ));
+    };
+
+    let aliases = if let ImportNames::Aliases(aliases) = &mut body.names {
+        aliases
+    } else {
+        return Err(anyhow::anyhow!("Expected node to be: Aliases"));
+    };
+
+    // Preserve the trailing comma (or not) from the last entry.
+    let trailing_comma = aliases.last().and_then(|alias| alias.comma.clone());
+
+    let removable = names
+        .iter()
+        .enumerate()
+        .filter_map(|(index, alias)| {
+            let name = &alias.node.name.as_str();
+            if (version >= PythonVersion::Py33 && PY33_PLUS_REMOVE_FUTURES.contains(name))
+                || (version >= PythonVersion::Py37 && PY37_PLUS_REMOVE_FUTURES.contains(name))
+            {
+                Some(index)
+            } else {
+                None
+            }
+        })
+        .collect::<Vec<_>>();
+
+    // TODO(charlie): This is quadratic.
+    for index in removable.iter().rev() {
+        aliases.remove(*index);
+    }
+
+    if let Some(alias) = aliases.last_mut() {
+        alias.comma = trailing_comma;
+    }
+
+    if aliases.is_empty() {
+        autofix::helpers::remove_stmt(stmt, None, &[])
```

Comment:
We do need to pass in `parent` and `deleted` here, as in `remove_unused_import_froms`.

Reply:
Got the `deleted`, I am at lost about how to get the `parent`

Reply:
All good, I’ll take a look and fix before merging :)